AsteroidOS/meta-smartwatch
# [DRAFT] Initial `rubyfish` port

(Sorry, I've been dragging my feet on this for a while)

This is the initial port to `rubyfish`. It still needs some rebasing, of the kernel parts in particular, but I'd like to request an initial review and any hints or advice as to how to improve the PR so it can be accepted.

Status of the port:
* kernel boots
* adbd over USB works
* psplash works
* asteroid-launcher works
  * performance seems acceptable
* touchscreen works
* battery indication works
* sensors
  * ALS seems miscalibrated (screen is too dark to see in dark rooms)
  * heart rate works
  * SpO2 works using `/vendor/bin/nanotool`, no AsteroidOS integration yet?
  * barometer works using `/vendor/bin/nanotool`, no AsteroidOS integration yet?
  * others are untested
* NFC is untested
* GPS is untested
* Bluetooth works
* WiFi works, but appears to scan with a random MAC address which means the connection will not be reused across reboots
* Audio does not work
  * it is possible to get audio output but this requires manual (command-line) initialization of the adsp and aw88230 i2c-controlled i2s amp
  * mic hasn't been tested yet. 
* secondary LCD display requires manual (command-line) initialization
* tilt-to-wake untested
* tap-to-wake untested
* AoD (non-LCD) doesn't update (same issue as `hoki`?)
* vibrator used to work, but no longer does...

TODO:
* renumber/rework Linux patches
* remove obsolete device trees etc. from Linux directory
* test on more than one watch
* port to `rover`
* fix AoD
